src/cross_section.F90

```diff
@@ -526,6 +572,185 @@ pure function find_energy_index(mat, E) result(i)
   end function find_energy_index
 
 !===============================================================================
+! MULTIPOLE_EVAL_ALLOCATE allocates fixed-length arrays that vary based on
+! what nuclides are loaded into the problem
+!===============================================================================
+
+  subroutine multipole_eval_allocate()
+    allocate(sigT_factor(max_L))
+    allocate(twophi(max_L))
+    allocate(broadened_polynomials(max_poly))
+
+    mp_already_alloc = .TRUE.
+  end subroutine
+
+!===============================================================================
+! MULTIPOLE_EVAL evaluates the windowed multipole equations for cross
+! sections in the resolved resonance regions
+!===============================================================================
+
+  subroutine multipole_eval(multipole, Emev, sqrtkT, sigT, sigA, sigF)
+    type(MultipoleArray), intent(in) :: multipole                    !< The windowed multipole object to process.
+    real(8), intent(in)              :: Emev                         !< The energy at which to evaluate the cross section in MeV
+    real(8), intent(in)              :: sqrtkT                       !< The temperature in the form sqrt(kT (in eV)), at which to evaluate the cross section.
+    real(8), intent(out)             :: sigT                         !< Total cross section
+    real(8), intent(out)             :: sigA                         !< Absorption cross section
+    real(8), intent(out)             :: sigF                         !< Fission cross section
+    complex(8) :: PSIIKI
+    complex(8) :: CDUM1
+    complex(8) :: w_val
+    complex(8) :: Z
+    real(8) :: sqrtE
+    real(8) :: invE
+    real(8) :: DOPP
+    real(8) :: DOPP_ECOEF
```

Comment:
We generally reserve all caps for `parameter` variables, so I'd suggest making these lowercase.